### DISCUSSION TITLE
Prisma: "Connect requires at least one argument" ... bug?

### DISCUSSION THREAD

**Discussion by florianmartens** [prisma/prisma#18391] [Q&A]
The following code caused a production bug, but ran through the typescript compiler without any warning:

```javascript
[...] this.prisma.share.create({
            data: {
                socialIdentity: {
                    connect: {
                        id: socialIdentityId || undefined,
                    },
                },
            },
        });
```

model:
```
model Share {
  id               String          @id @default(cuid())
  socialIdentityId String?
  socialIdentity   SocialIdentity? @relation(fields: [socialIdentityId], references: [id], onDelete: SetNull)
}
```

The error states:
`Argument data.socialIdentity.connect of type SocialIdentityWhereUniqueInput needs at least one argument. Available args are listed in green.`

I'm not sure what went wrong. Of course my intention was to connect a record `socialIdentity` if it exists and to connect no record if none exists. Why is prisma not able to handle this correctly? I'm on version `4.10.1`.

**Comment by ludralph** [comment]
Hi @florianmartens 👋 

What happens if you directly set the `socialIdentityId` value?
```js
const share = await prisma.share.create({
  data: {
    socialIdentityId: socialIdentityId || undefined
  },
});
```

**Reply by ludralph** [reply]
Hey @florianmartens ,

It would be really helpful if you could provide the additional information that we had asked about last time. If your problem is resolved by now or somehow doesn't feel relevant any more, you can simply close the Discussion as well.

**Reply by florianmartens** [reply]
Ah, it shows a type error when I want to specify the socialIdentityId directly.

**Reply by ludralph** [reply]
Is it possible to share a minimal reproduction of this issue in a git repository so i can take a look?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
